mpr-research-data.py

```diff
@@ -145,140 +149,72 @@ def set(self, name, value):
-            if not os.path.isfile(os.path.join(self.defaultQueryFolder, self.queryTemplateDict[queryType])):
-                logging.error(
-                    f'SQL Query file for {queryType} not found in query directory {self.defaultQueryFolder}.')
-                return False
-            return True
-
-        except TypeError as e:
-            logging.error(f'Error Message: {e}')
-            logging.error(
-                f'Invalid type (expected str) passed for query type: {queryType}.')
-            return False
-
-    def setAndVerifyDBCreds(self):
-
-        dbCredsDefaultDict = {'NAME': None,
-                              'USER': None,
-                              'PASSWORD': None,
-                              'HOST': None,
-                              'PORT': 3306}
-        self.dbParams = {}
-
-        allKeyPartsFound = True
-        for credPart in dbCredsDefaultDict:
-
-            if credPart == 'PORT':
-                try:
-                    self.dbParams[credPart] = int(os.getenv(
-                        'DB_' + credPart, dbCredsDefaultDict[credPart]))
-                except ValueError as e:
-                    logging.error(f'Error Message: {e}')
-                    logging.error(
-                        f'Invalid type (expected int) passed for DB_{credPart}.')
-                    allKeyPartsFound = False
-
-            else:
-                try:
-                    self.dbParams[credPart] = str(os.getenv(
-                        'DB_' + credPart, dbCredsDefaultDict[credPart]))
-                except TypeError as e:
-                    logging.error(f'Error Message: {e}')
-                    logging.error(
-                        f'Invalid type (expected str) passed for DB_{credPart}.')
-                    allKeyPartsFound = False
-
-            if not self.dbParams.get(credPart, False):
-                logging.error(
-                    f'Did not find configuration parameter for M-Write Peer Review production DB key: {credPart}.')
-                allKeyPartsFound = False
-
-        return allKeyPartsFound
-
-    def setAndVerifyGCPCreds(self):
-        try:
-            self.gcpParams = json.loads(str(os.getenv('GCP_KEY')))
-            return True
-        except json.JSONDecodeError as e:
-            logging.error(f'Error Message: {e}')
-            logging.error(
-                f'Invalid JSON format passed for GCloud Service JSON Key.')
-            return False
+    def configFetch(self, name, default=None, casting=None, validation=None, valErrorMsg=None):
+        value = os.getenv(name, default)
+        if (casting is not None):
+            try:
+                value = casting(value)
+            except:
+                errorMsg = f'Casting error for config item "{name}" value "{value}".'
+                logging.error(errorMsg)
+                return None
+                #raise TypeError(errorMsg)
```

Comment:
So, for this commented `raise` line and the one a few lines down, what's your thinking?  Do you feel more comfortable handling the `None` return values than catching exceptions?  If you think that's the way to go, you should definitely delete the commented lines.

Reply:
I like to handle Nones better as I do check if all config values have been set up correctly, even if one already failed, as a way to find all errors on file in one go. I will remove the commented lines.

Reply:
OK.  `None` it is, then.  I'll watch for the comment removal.

Reply:
✅